### PR TITLE
port text_viewer from gtk-rs/examples4

### PR DESCRIPTION
**Pull request by vamsikrishna-brahmajosyula**

### CODE REVIEW

File: examples/src/bin/ui/text_viewer.ui [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+            </child>
+          </object>
+        </child>
+        <child>
+          <object class="GtkScrolledWindow" id="scrolled_window">
+            <property name="hexpand">True</property>
+            <property name="vexpand">True</property>
+            <child>
+              <object class="GtkTextView" id="text_view">
+              </object>
```
Comment (bilelmoussaoui) [inline examples/src/bin/ui/text_viewer.ui:27]:
```suggestion
              <object class="GtkTextView" id="text_view" />
```

File: examples/src/bin/text_viewer.rs [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+    let open_button: Button = builder
+        .get_object("open_button")
+        .expect("Couldn't get builder");
+    let text_view: TextView = builder
+        .get_object("text_view")
+        .expect("Couldn't get text_view");
+
+    open_button.connect_clicked(glib::clone!(@weak window, @weak text_view => move |_| {
+
+        // TODO move this to a impl?
```
Comment (bilelmoussaoui) [inline examples/src/bin/text_viewer.rs:36]: I think it's fine to keep it here for this example, so please remove the todo

**Comment by bilelmoussaoui** [comment]
Looks good to me otherwise, thanks!

**Comment by sdroege** [comment]
Can you squash "Address review comments" commits into the corresponding commit in the future? Thanks :)

**Comment by bilelmoussaoui** [comment]
> Can you squash "Address review comments" commits into the corresponding commit in the future? Thanks :)

I did that when I merged?

**Comment by sdroege** [comment]
Ah sure, that also works :) IMHO it's less error-prone to do that before merging with git

**Pull request merged**